### ISSUE TITLE
Inconsistency between IO.inspect :limit and :printable_limit

### ISSUE THREAD

**Issue by vans163** [elixir-lang/elixir#9108]
Elixir 1.8.2

So I really hope this does not get marked as a documentation issue but there is an inconsistency between :limit and :printable_limit, the docs say

```
:limit - limits the number of items that are printed for tuples, bitstrings, maps, lists and any other collection of items. It does not apply to strings nor charlists and defaults to 50.

:printable_limit - limits the number of bytes that are printed for strings and charlists. 
```

But if you just do IO.inspect(a_binary, printable_limit: 10000), without :limit, it does not apply.  You must set both IO.inspect(a_binary, limit: 10000, printable_limit: 10000).

Example:
```
iex(5)> IO.inspect :binary.copy(<<1>>,100000), printable_limit: 10000
<<1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...>>

iex(6)> IO.inspect :binary.copy(<<1>>,100000), limit: 10000, printable_limit: 10000
<<1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
~she goes~
```

**Comment by josevalim** [comment]
A raw binary is handled under :limit. The printable_limit applies to
strings (which are UTF-8 encoded) and char lists. Therefore you need to set
only :limit in this example. A good way to think about it is that
printable_limit applies to things showed between double and single quotes.
-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

**Comment by vans163** [comment]
This is so confusing. Often working with strings/binaries/bitstrings/any_others? if I do not put both it doesn't print things, depends how elixir interprets the binary/string/bitstring/any_others? at moment of inception?

**Comment by josevalim** [comment]
Different representations have different rules on how they should be printed, there is nothing surprising about it. You also don't have to rely on what Elixir infers, you can ask Elixir to treat all binaries as strings or as raw binaries. All of the options are in place.

Since Elixir has to work out of the box on a huge variety of contexts, then our default behaviour has to be a guess. Otherwise, printing strings for someone implementing a binary protocol would be bad, and printing binaries for someone loading strings from a database would be bad too.

**Comment by vans163** [comment]
"you can ask Elixir to treat all binaries as strings or as raw binaries."

Omg how? I have been dreaming of this.  Is it a VM / config argument?  System wide? That would be so sweet.

Also to always print floats as they are, and not switch to e notation system wide would be a huge help.

**Comment by wojtekmach** [comment]
> "you can ask Elixir to treat all binaries as strings or as raw binaries."
> 
> Omg how? I have been dreaming of this. Is it a VM / config argument? System wide? That would be so sweet.
> 
You can set it via binaries: :as_strings and :as_binaries, respectively. See Inspect.Opts for docs. There’s no VM argument, the closest there is to global config is IEx.configure(inspect: ...)
> Also to always print floats as they are, and not switch to e notation system wide would be a huge help.
> 
You can overwrite how floats are displayed using the new :inspect_fun option available in v1.9.
